internal/app/frontend/frontend_service_test.go

```diff
@@ -340,6 +340,88 @@ func TestDoWatchAssignments(t *testing.T) {
 	}
 }
 
+// TestAcknowledgeBackfill - test input validation
+func TestAcknowledgeBackfillValidation(t *testing.T) {
+	cfg := viper.New()
+	tests := []struct {
+		description string
+		request     *pb.AcknowledgeBackfillRequest
+	}{
+		{
+			description: "check backfillId exists",
+			request:     &pb.AcknowledgeBackfillRequest{BackfillId: "", Assignment: &pb.Assignment{Connection: "10.0.0.1"}},
+		},
+		{
+			description: "check Assignment exist",
+			request:     &pb.AcknowledgeBackfillRequest{BackfillId: "1234", Assignment: nil},
+		},
+		{
+			description: "check Assignment.connection",
+			request:     &pb.AcknowledgeBackfillRequest{BackfillId: "1234", Assignment: &pb.Assignment{Connection: ""}},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.description, func(t *testing.T) {
+			ctx := context.Background()
+
+			store, closer := statestoreTesting.NewStoreServiceForTesting(t, cfg)
+			defer closer()
+			fs := frontendService{cfg, store}
+			bf, err := fs.AcknowledgeBackfill(ctx, test.request)
+			require.Equal(t, codes.InvalidArgument.String(), status.Convert(err).Code().String())
+			require.Nil(t, bf)
+		})
+	}
+}
+
+// TestAcknowledgeBackfill - CreateBackfill,
+func TestAcknowledgeBackfill(t *testing.T) {
+	cfg := viper.New()
+	ctx := context.Background()
+
+	store, closer := statestoreTesting.NewStoreServiceForTesting(t, cfg)
+	defer closer()
+
+	fakeBackfill := &pb.Backfill{
+		Id: "1",
+		SearchFields: &pb.SearchFields{
+			DoubleArgs: map[string]float64{
+				"test-arg": 1,
+			},
+		},
+	}
+	err := store.CreateBackfill(ctx, fakeBackfill, []string{})
+	require.NoError(t, err)
+	fs := frontendService{cfg, store}
+	ids, err := store.GetExpiredBackfillIDs(ctx)
+	require.NoError(t, err)
+	require.Len(t, ids, 0)
+
+	time.Sleep(cfg.GetDuration("pendingReleaseTimeout"))
+	ids, err = store.GetExpiredBackfillIDs(ctx)
+	require.NoError(t, err)
+	require.Len(t, ids, 1)
```

Comment:
these 2 calls of GetExpiredBackfillIDs can be safely removed because here you test GetExpiredBackfillIDs actually and they are not related to TestAcknowledgeBackfill. In this test, we assume that they work as expected, so please don't waste test execution time using sleep on these steps. This functionality is properly covered in statestore tests.